runtime/src/iree/base/internal/dynamic_library.h

```diff
@@ -73,6 +73,13 @@ iree_status_t iree_dynamic_library_attach_symbols_from_file(
 iree_status_t iree_dynamic_library_attach_symbols_from_memory(
     iree_dynamic_library_t* library, iree_const_byte_span_t buffer);
 
+// Queries the operating system for the for the full path of the dynamic
+// library in which |for_symbol| is located. For symbols not in a dynamic
+// library, the behavior is implementation dependent (i.e. it may return
+// a failure status or it may return a path to an executable, etc).
+iree_status_t iree_dynamic_library_get_symbol_path(
+    void* for_symbol, iree_string_builder_t* out_path);
```

Comment:
```suggestion
iree_status_t iree_dynamic_library_append_symbol_path_to_builder(
    void* for_symbol, iree_string_builder_t* builder);
```
(out_ implies that it's overwriting, whereas this is appending - matches `iree_hal_buffer_view_append_to_builder`)